### PR TITLE
updated email address and author in setup.py files

### LINKED ISSUE #3174: change author email address to official MDA email
## Expected behavior ##

I would expect an official MDA email like mdanalysis@numfocus.org to be listed in `{package,testsuite}/setup.py` 

## Actual behavior ##

For historical reasons, the author email address in setup.py is still @nmichaud 's old address. 
https://github.com/MDAnalysis/mdanalysis/blob/6282385644d672cf7023584347e35048935ab429/package/setup.py#L600 and https://github.com/MDAnalysis/mdanalysis/blob/6282385644d672cf7023584347e35048935ab429/testsuite/setup.py#L120

However, I doubt that he's too keen to field MDA support inquiries...

NOTE: This is a policy issue and **not eligible for GSoC PRs** — only PRs from @MDAnalysis/coredevs will be reviewed.

### PR DESCRIPTION
**Pull request by orbeckst**
Fixes #3174 

Changes made in this Pull Request:
- use mdanalysis@numfocus.org as official address
- use MDAnalysis Development Team as author
- close  #3174

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/setup.py
Comment (orbeckst) [inline package/setup.py:601]: Are we happy with _MDAnalysis Development Team_ as author?

File: package/setup.py
Comment (orbeckst) [inline package/setup.py:602]: Is @IAlibay current maintainer or should this still be @richardjgowers ?

**Comment by orbeckst** [comment]
@MDAnalysis/coredevs are you happy with this change? Pinging everyone for authorship changes...

I'd appreciate a review from @nmichaud .

File: package/setup.py
Comment (lilyminium) [reply package/setup.py:601]: Yup, I'm happy with it. I'll let @richardjgowers / @IAlibay / @nmichaud approve the PR though.

File: package/setup.py
Comment (IAlibay) [reply package/setup.py:602]: @richardjgowers still holds all the keys, I'd like to think of myself as the sith apprentice 🙃 

That being said, if we're looking to change it, shouldn't the maintainers be "The MDAnalysis Core Developer Team"?

**Review by IAlibay — COMMENTED** [review]
lgtm, just the one comment

File: package/setup.py
Comment (orbeckst) [reply package/setup.py:602]: I like your suggestion. At least then blame gets properly apportioned.

@richardjgowers what's your opinion?

Comment (tylerjereddy) [reply package/setup.py:602]: Just for another data point--SciPy mentions the person holding the release manager position at a given time in the online docs, but not in the source as far as I know: https://scipy.github.io/devdocs/dev/governance/people.html?highlight=tyler%20reddy#release-manager

maintainers is "SciPy Developers"

**Review by tylerjereddy — APPROVED** [review]
LGTM -- if Irfan and Richard are still doing some heavy lifting for releases I'd be +1 for their roles to be recognized in the docs somewhere. I can't remember if we have a kind of "roles" section.

We also probably have a fairly clear candidate for the person who leads doc maintenance too, right? I guess maintaining those kinds of lists gets a bit awkward when people shift around a bit, although you can just add an "emeritus" section to be a bit more graceful and say you'll welcome people back with open arms when you do make PRs to change them.

**Comment by nmichaud** [comment]
I volunteer to be the first member of the emeritus section :)

Naveen

**Comment by orbeckst** [comment]
ping @nmichaud @richardjgowers for a quick review, please — your names are getting changed so it would be good to have your official approval for the change. Thanks.

**Review by richardjgowers — APPROVED** [review]
Yeah I didn't even realise I was personally listed. I think maintainer as the collective is better

File: package/setup.py
Comment (orbeckst) [reply package/setup.py:602]: In response to @IAlibay and @richardjgowers comments I am changing the maintainer name to the official title [MDAnalysis Core Developers](https://www.mdanalysis.org/about/#governance). 

(I was striving for grammatical uniformity but If you prefer the article "The" in front I can be swayed.)

**Comment by orbeckst** [comment]
I changed the maintainer name. 

I also updated metadata links and added a few new ones to show up in the pypi page
- User Guide
- Blog
- Twitter
- Discord (this one is a bit iffy because you first need the invite but posting the invite link there will invite again people who are already on our discord as opposed to just letting them go to the discord server)

**Review by nmichaud — APPROVED** [review]

**Review by nmichaud — APPROVED** [review]
Looks good to me.

**Pull request merged**
